var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -995,6 +1005,12 @@ def configure_args(self):
             self.enable_or_disable("mpi-thread-multiple", variant="thread_multiple")
         )
 
+        # With OpenMPI 4.0 and above, there could be compilation errors from “btl_uct” component.
+        # This component is not critical for using UCX; so it could be disabled.
+        # See https://openucx.readthedocs.io/en/master/running.html#id1
+        if "^ucx" in spec and spec.satisfies("@4:"):
+            config_args.append("--enable-mca-no-build=btl-uct")
+
```

Comment:
Did you actually run into an issue?

I didn't read the full thread, but looks like it was resolved: https://github.com/open-mpi/ompi/issues/6666